docs/design.md

```diff
@@ -1,175 +1,250 @@
 # Design
 
-## Models
+## Common Functionality across all Kinds
 
-A Model object represents a logical ML model. The object describes a location
-where ML model files can be stored for a specific model. 
+### Container Images
 
-Optionally, a loader can be specified to load ML models weights and biases from specific sources.
-For example, a commonly used model loader would be the HuggingFace model loader
-where by providing the model name would load the model to the Model object
-bucket location. Model loaders are defined as container images and can either
-point to prebuilt container images or a Git repository with Dockerfile.
+All Substratus kinds correspond to actions that are taken by containers. These containers can be built
+by Substratus (from git, or from an upload) or referenced from external registries.
 
-A model
+```yaml
+spec:
+  image:
+    # Optional git source.
+    git:
+      url: https://github.com/substratusai/hf-model-loader
+    
+    # Optional upload source.
+    upload: {}
+
+    # Optionally specify image name to externally published container image
+    # This field will be set by the controller if a build-source (above) was used
+    # and should not be set by end-user if a build-source was used.
+    name: substratusai/hf-model-loader
```

Comment:
Shouldn't these be directly under `spec.` instead of `spec.image.`?
```suggestion
  upload: {}

  # Optionally specify image name to externally published container image
  # This field will be set by the controller if a build-source (above) was used
  # and should not be set by end-user if a build-source was used.
  name: substratusai/hf-model-loader
```